torchgeo/datasets/seasonet.py

```diff
@@ -0,0 +1,456 @@
+        "Port areas",
+        "Airports",
+        "Mineral extraction sites",
+        "Dump sites",
+        "Construction sites",
+        "Green urban areas",
+        "Sport and leisure facilities",
+        "Non-irrigated arable land",
+        "Vineyards",
+        "Fruit trees and berry plantations",
+        "Pastures",
+        "Broad-leaved forest",
+        "Coniferous forest",
+        "Mixed forest",
+        "Natural grasslands",
+        "Moors and heathland",
+        "Transitional woodland/shrub",
+        "Beaches, dunes, sands",
+        "Bare rock",
+        "Sparsely vegetated areas",
+        "Inland marshes",
+        "Peat bogs",
+        "Salt marshes",
+        "Intertidal flats",
+        "Water courses",
+        "Water bodies",
+        "Coastal lagoons",
+        "Estuaries",
+        "Sea and ocean",
+    ]
+    all_seasons = {"Spring", "Summer", "Fall", "Winter", "Snow"}
+    all_bands = ("10m_RGB", "10m_IR", "20m", "60m")
+    band_nums = {"10m_RGB": 3, "10m_IR": 1, "20m": 6, "60m": 2}
+    splits = ["train", "val", "test"]
+    colormap = [
+        (230, 000, 77, 255),
+        (255, 000, 000, 255),
+        (204, 77, 242, 255),
+        (204, 000, 000, 255),
+        (230, 204, 204, 255),
+        (230, 204, 230, 255),
+        (166, 000, 204, 255),
+        (166, 77, 000, 255),
+        (255, 77, 255, 255),
+        (255, 166, 255, 255),
+        (255, 230, 255, 255),
+        (255, 255, 168, 255),
+        (230, 128, 000, 255),
+        (242, 166, 77, 255),
+        (230, 230, 77, 255),
+        (128, 255, 000, 255),
+        (000, 166, 000, 255),
+        (77, 255, 000, 255),
+        (204, 242, 77, 255),
+        (166, 255, 128, 255),
+        (166, 242, 000, 255),
+        (230, 230, 230, 255),
+        (204, 204, 204, 255),
+        (204, 255, 204, 255),
+        (166, 166, 255, 255),
+        (77, 77, 255, 255),
+        (204, 204, 255, 255),
+        (166, 166, 230, 255),
+        (000, 204, 242, 255),
+        (128, 242, 230, 255),
+        (000, 255, 166, 255),
+        (166, 255, 230, 255),
+        (230, 242, 255, 255),
+    ]
+    image_size = (120, 120)
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "train",
+        seasons: Collection[str] = all_seasons,
+        bands: Iterable[str] = all_bands,
+        grids: Iterable[int] = [1, 2],
+        concat_seasons: int = 1,
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new SeasoNet dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train", "val" or "test"
+            seasons: list of seasons to load
+            bands: list of bands to load
+            grids: which of the overlapping grids to load
+            concat_seasons: number of seasonal images to return per sample.
```

Comment:
The SeCo and SSL4EO datasets also have a notion of seasons. In those datasets, the `seasons` parameter controls the number of seasons to sample from. I wonder if we should try to be more consistent with parameter names or not.

Reply:
What `seasons` does in both of those datasets is basically what `concat_seasons` does in our dataset. I do agree with trying to be consistent but I also think it would be a bit counter-intuitive for the `seasons` parameter not to be the one that is used to choose the actual seasons. Though it would require more changes, I could rather see something like `num_seasons` working in all cases.